homeassistant/components/flux_led/config_flow.py

```diff
@@ -132,6 +132,8 @@ async def async_step_prompt_options(self, user_input=None):
                 config_data = self._config_entry.data.copy()
                 del config_data[CONF_DEVICES][device_id]
 
+                self.hass.config_entries.async_update_entry(self._config_entry, data={})
```

Comment:
Why update config entry data twice in a row?

Reply:
For some reason the `async_update_entry` doesn't detect the change in data when it is sent through the first time. So I am clearing the data, then rewriting it and that forces the `async_update_entry` to see the change and actually save it to `config_entries`. Without this double call every time HA restarts it brings back any devices you removed and any devices you added aren't there to be managed.

Reply:
Then something isn't working as it should. We need to find what that is and not add a work around.

Reply:
I have validated by capturing in logs that the data being sent to async_update_entry does not match the data in the config_entries, but the core.config_entries file doesn't get updated unless I run this double send. I also tried a copy.deepcopy to try to copy the dict in more detail instead of dict.copy() but because it is a Mapping object that isn't supported.

Any other suggestions?

Reply:
I haven't found it yet. Did this work before this PR?

Reply:
I went back to the last PR version to test once I noticed the behaviour and it was the same. I'm not sure if it is because the changes are a level deep in the dict?

Reply:
Yeah, I think that's it. `.copy` only returns a shallow copy. Probably create a new dict and and populate it with the entry data instead.

Reply:
Is it more efficient to parse the existing config_data into a new dict then call the update or just call the update twice as I'm doing?